rfc100-resource-lifecycle.md

```diff
@@ -63,56 +88,54 @@ Resources in retired state have been in deprecated state and emitting warnings a
-* well tested and supported cookbook resources,
-* percentage of supported platforms,
-* impact of implementing as a primitive that can be built upon.
+- widespread use of a resource based on analysis of publicly available supermarket cookbooks,
+- well tested and supported cookbook resources,
+- percentage of supported platforms,
+- impact of implementing as a primitive that can be built upon.
 
 ### Criteria for deprecating resources
 
 Factors that influence and inform the decision to deprecate a resource include:
 
-* limited use based on analysis of publicly available supermarket cookbooks,
-* outdated code or lack of support based on quality of resource and its usability,
-* security implications,
-* complex implementation that lacks clarity of consequences of use.
-* impact of using the resource incorrectly.
+- limited use based on analysis of publicly available supermarket cookbooks,
+- outdated code or lack of support based on quality of resource and its usability,
+- security implications,
+- complex implementation that lacks clarity of consequences of use.
+- impact of using the resource incorrectly.
+
+### Process for proposing / developing new resources
+
+Users may propose new resources for inclusion in the chef-client by filing a resource proposal issue in the Chef GitHub repository. The chef-client maintainers will work with the user proposing the new resource to fully document the needs and the proposed design. Upon approval from two core maintainers the proposal will be marked `Status: Approved for Development` and the development phase may begin.
 
-### Process for adopting resources from a community cookbook
+#### Proposal Process
 
-* Identify resources for adoption.
-* Create an RFC following the [Chef RFC](https://github.com/chef/chef-rfc) process.
-* Announce on Chef mailing list and Chef Community Slack.
-* Any proposed changes to the interfaces must be implemented in the cookbook prior to adoption.
-  * If there are fundamental issues with the resource whether due to naming standards or interface implementation, a new cookbook should be created implementing the resource as intended.
-* Any bugs discovered must be repaired in the cookbook and released prior to adoption.
-* Resource is added to core chef and documentation is updated.
-* Add deprecated cookbook warnings for conflicting cookbooks.
+- Identify resources for adoption.
+- Create an issue at [https://github.com/chef/chef] outlining the reason for shipping the resource in core.
+- Announce your proposal in the #chef-dev channel on Chef Community Slack.
+- Chef-client maintainers will work with you to fully document the needs and the proposed design of the resource.
+- Once two chef-client maintainers have approved the proposal the `Status: Approved for Development` label will be applied and development may begin.
+
+#### Development Process
+
+- All development must take place initially in a cookbook.
```

Comment:
I strongly object to this. While this is **often** the best case, it's not **always** the case. Adding a new `package` resource, say, **dnf_package** should **not** have happened in a cookbook first, it's nonsensicle. I'm fine ENCOURAGING this, but it absolutely is NOT a "must" - I fought against it being in the original RFC, and i will fight against it being in this one too.

Reply:
@jaymzh I think I appreciate where you're coming from, but I prefer the idea of having dev take place in a cookbook first for the fact that you aren't locked into the chef-client release cycles which are slow (for a good reason). This especially bites you when you require changing behavior in a non-backwards compatible way because of the way behavior MUST be slowly changed once introduced in the client core.

Reply:
I disagree. Adding DNF support to `package` should *not* happen in a cookbook. I upgrade my OS, and now you tell me I have to get a cookbook for Chef to work right? No, that's absurd, I should be able to grab the latest chef and move on. There are a lot of resources that would be nice and core and it's unclear if they'll make it into core so they need incubation inside of a cookbook, but there are many that *must* be in core. Supporting the new package format that Fedora moves to, or updating `service` to support systemd, for example, are non-optional. If we didn't support systemd for service, Chef would have ceased to be able to be used on any modern Linux distro. Putting that in a cookbook doesn't make sense, it should be developed and on and release with core Chef because these are core parts of Chef.

It's one thing to say the default path is through cookbooks, but it's a **whole separate** thing to codify a rule that no Chef developer - employee or otherwise - can submit a PR to add systemd support to the `service` provide without publishing a cookbook first. That's not reasonable.